### PR TITLE
feat(5-breadcrumbs): add breadcrumbs to plp

### PR DESCRIPTION
**Pull request by stephenrichter**
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 13 05 33

### CODE REVIEW

File: lib/flex_ui/widgets/search_results_header.dart [hunk 1]
```diff
@@ -1,54 +1,101 @@
...
+          ),
         Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
           children: [
-            TextButton.icon(
-              onPressed: () async {
-                final result = await context.router.root
-                    .push<Sort?>(SortRoute(sorts: searchResults.sorts));
+            Text('${searchResults.pagination.totalResults} items'),
+            Row(
```
Comment (ncuillery) [inline lib/flex_ui/widgets/search_results_header.dart]: Can you please add a screenshot in the PR description? 

I'm suspecting too much nested Rows but I'm not sure by just looking at the code.

File: lib/search/models/search_results.dart [hunk 2]
```diff
@@ -27,7 +29,34 @@ class SearchResults {
...
+}
+
+@JsonSerializable(createToJson: false)
+class Breadcrumb {
+  final String? facetCode;
+  final String? facetName;
+  final String? facetValueCode;
+  final String? facetValueName;
+  final SearchState? removeQuery;
+  final SearchState? truncateQuery;
```
Comment (ncuillery) [inline lib/search/models/search_results.dart:43]: Some of those fields can be declared as non-nullable, can't they?

File: lib/flex_ui/widgets/search_results_header.dart [hunk 3]
```diff
@@ -1,54 +1,101 @@
...
   @override
   Widget build(BuildContext context) {
-    return Row(
-      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        Text('${searchResults.pagination.totalResults} items'),
+        if (breadcrumbs.isNotEmpty)
+          BreadCrumb(
```
Comment (ncuillery) [inline lib/flex_ui/widgets/search_results_header.dart:25]: Hmmm widget named `BreadCrumb`, model named `Breadcrumb`... there should be a better way 😄  (but I don't find any on top of my head)

Comment (stephenrichter) [reply lib/flex_ui/widgets/search_results_header.dart:25]: Normally I would change it, but they are direct from Package and from Hybris. Better to keep the naming consistent with the source in my opinion.

**Pull request merged**